### PR TITLE
Remove Compass references from Provisioner component

### PR DESCRIPTION
**Pull request by Maladie**
**Description**

Changes proposed in this pull request:

- references removed from component, components tests, docs and charts

**Related issue(s)**
Issue: https://github.com/kyma-project/control-plane/issues/38

### CODE REVIEW

File: components/provisioner/cmd/main.go [hunk 1]
```diff
@@ -59,7 +59,7 @@ type config struct {
 	CredentialsNamespace         string `envconfig:"default=compass-system"`
```
Comment (franpog859) [inline components/provisioner/cmd/main.go:59]: The default value could be changed to `kcp-system`

**Comment by rafalpotempa** [comment]
@franpog859 Yes, it failed on deprovisioning timeout, but the configs on provisioned runtimes were verified successfully.

**Comment by franpog859** [comment]
@majakurcius , I think we need your approve here 😅

File: components/provisioner/README.md [hunk 2]
```diff
@@ -43,10 +43,10 @@ This table lists the environment variables, their descriptions, and default valu
 | **APP_ADDRESS** | Provisioner address with the port | `127.0.0.1:3000` |
```
Comment (majakurcius) [inline components/provisioner/README.md:43]:
```suggestion
| **APP_ADDRESS** | Runtime Provisioner's address with the port | `127.0.0.1:3000` |
```

File: components/provisioner/README.md [hunk 3]
```diff
@@ -43,10 +43,10 @@ This table lists the environment variables, their descriptions, and default valu
...
 | **APP_API_ENDPOINT** | Endpoint for the GraphQL API | `/graphql` |
 | **APP_PLAYGROUND_API_ENDPOINT** | Endpoint for the API playground | `/graphql` |
-| **APP_CREDENTIALS_NAMESPACE** | Namespace where Director credentials are stored | `compass-system` |
 | **APP_DIRECTOR_URL** | Director URL | `https://compass-gateway-auth-oauth.kyma.local/director/graphql` |
 | **APP_SKIP_DIRECTOR_CERT_VERIFICATION** | Flag to skip certificate verification for Director | `false` |
-| **APP_OAUTH_CREDENTIALS_SECRET_NAME** | Runtime Provisioner credentials | `compass-provisioner-credentials` |
+| **APP_OAUTH_CREDENTIALS_NAMESPACE** | Namespace where Director credentials are stored | `kcp-system` |
```
Comment (majakurcius) [inline components/provisioner/README.md:48]:
```suggestion
| **APP_OAUTH_CREDENTIALS_NAMESPACE** | Namespace where the Director credentials are stored | `kcp-system` |
```

File: docs/provisioner/01-01-provisioner-overview.md [hunk 4]
```diff
@@ -3,7 +3,7 @@ title: Overview
 type: Overview
 ---
 
-The Runtime Provisioner is a Compass component responsible for provisioning, installing, and deprovisioning clusters with Kyma (Kyma Runtimes). The relationship between clusters and Runtimes is 1:1. The Runtime Provisioner is registered in the Director as an Integration System.
+The Runtime Provisioner is a Kyma Control Plane component responsible for provisioning, installing, and deprovisioning clusters with Kyma (Kyma Runtimes). The relationship between clusters and Runtimes is 1:1. The Runtime Provisioner is registered in the Compass Director as an Integration System.
```
Comment (majakurcius) [inline docs/provisioner/01-01-provisioner-overview.md:6]:
```suggestion
The Runtime Provisioner is a Kyma Control Plane component responsible for provisioning, installing, and deprovisioning clusters with Kyma (Kyma Runtimes). The relationship between clusters and Runtimes is 1:1. The Runtime Provisioner is registered in Compass in the Director as an Integration System.
```
The name of the component is Director, and that's how we refer to it in other places, too.

File: components/provisioner/internal/gardener/annotations.go [hunk 5]
```diff
@@ -21,8 +21,8 @@ func (s KymaInstallationState) String() string {
 }
 
 const (
-	operationIdAnnotation string = "compass.provisioner.kyma-project.io/operation-id"
-	runtimeIdAnnotation   string = "compass.provisioner.kyma-project.io/runtime-id"
+	operationIdAnnotation string = "kcp.provisioner.kyma-project.io/operation-id"
+	runtimeIdAnnotation   string = "kcp.provisioner.kyma-project.io/runtime-id"
```
Comment (Szymongib) [inline components/provisioner/internal/gardener/annotations.go:25]: I am not sure if we can just change these annotations as they are used by other teams. I suggest introducing the second set of annotations starting with `kcp` and inform others about the change to give them time to switch.

**Pull request merged**